### ISSUE TITLE
Formatter moves `noqa: PLR0913`, causing false `RUF100` and unsuppressed `PLR0913`

### ISSUE THREAD

**Issue by SerTetora** [astral-sh/ruff#23332]
### Summary

When formatting a function with a long signature and an inline `# noqa: PLR0913`, Ruff reformats the signature across multiple lines and leaves the `noqa` on the closing parenthesis line.

Input:

```python
def test(first_argument, second_argument, third_argument, fourth_argument, fifth_argument, sixth_argument, seventh_argumement) -> None:  # noqa: PLR0913
    pass
```

After formatting:

```python
def test(
    first_argument, second_argument, third_argument, fourth_argument, fifth_argument, sixth_argument, seventh_argumement
) -> None:  # noqa: PLR0913
    pass
```

Diagnostics:

```
Too many arguments in function definition (7 > 5) (PLR0913)
Unused `noqa` directive (unused: `PLR0913`) (RUF100)
``` 

https://play.ruff.rs/133a6921-166e-4d3d-80b1-589583d2a933

### Version

15.1

**Comment by ntBre** [comment]
Thank you for the report! Unfortunately this is a known issue that is tricky to resolve. I'll close this as a duplicate of #12901 to keep discussion in one place.
